### PR TITLE
Add Authentik documentation page and setup instructions

### PR DESCRIPTION
**Pull request by RaneyDazed**
This pull request adds a new documentation page for Authentik and updates the setup instructions and user login stage configuration. It also includes additional email settings for Authentik.

### CODE REVIEW

File: docs/apps/authentik.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+``` shell
+
+sb install authentik
+
+```
+
+### 2. URL
+
+- To access Authentik, visit `https://authentik._yourdomain.com_`
```
Comment (owine) [inline docs/apps/authentik.md:28]: This is not the correct default subdomain - it's auth.

Comment (RaneyDazed) [reply docs/apps/authentik.md:28]: Omg I know why didn't I change that. I even made note of it. Thanks!

**Comment by RaneyDazed** [comment]
> Instructions on how to use Authentik for one specific role or to override and replace Authelia globally would probably be useful on this page as well

10-4. I thought about doing both of those things, but wasn't sure if it was something you guys would like me to add. I'll do that now.

File: docs/apps/authentik.md [hunk 2]
```diff
@@ -39,7 +41,23 @@ sb install authentik
...
+
+To enable Authentik globally, set the following [inventory](../saltbox/inventory/index.md) entry:
+
+``` yaml title="Authentik Global Toggle"
+
+traefik_default_sso_middleware: "authentik@docker"
+
+```
+
+You would use one or the other, the global toggle will install Authentik together with Traefik when it is being installed, the role toggle requires manually installing Authentik. Traefik needs to be reinstalled at least once after updating to deploy the Authentik middleware to enable use of it. If you are using the global toggle, you will need to reinstall all of the other roles you use to enable Authentik.
```
Comment (owine) [inline docs/apps/authentik.md:60]: The global toggle will not result in the `authentik` role running when the `traefik` role runs. They will always need to be run as separate tags.

**Review by owine — CHANGES_REQUESTED** [review]

File: docs/apps/authentik.md (hunk 2 above)
Comment (saltydk) [reply docs/apps/authentik.md:60]: It does actually :p

Comment (saltydk) [reply docs/apps/authentik.md:60]: https://github.com/saltyorg/Saltbox/blob/80d3c3e13b4538dd37330e3e0bc310c6e469de92/roles/traefik/tasks/main.yml#L90-L93

Comment (RaneyDazed) [reply docs/apps/authentik.md:60]: For what its worth, I used saltys post in discord for that. It was copy and pasted from his announcement lmao

Comment (RaneyDazed) [reply docs/apps/authentik.md:60]: I'm adding an oidc example to the docs though, so not quite done yet.

**Comment by saltydk** [comment]
@owine @RaneyDazed this was merged a bit too early I think, the oidc page isn't referenced at all?

**Pull request merged**